mesh/mesh.go

```diff
@@ -96,6 +102,38 @@ func NewMesh(db *MeshDB, atxDb AtxDB, rewardConfig Config, mesh MeshValidator, t
 	return ll
 }
 
+func NewRecoveredMesh(db *MeshDB, atxDb AtxDB, rewardConfig Config, mesh MeshValidator, txInvalidator TxMemPoolInValidator, atxInvalidator AtxMemPoolInValidator, pr TxProcessor, logger log.Log) *Mesh {
+	//todo add boot from disk
+	ll := &Mesh{
+		Log:            logger,
+		MeshValidator:  mesh,
+		txInvalidator:  txInvalidator,
+		atxInvalidator: atxInvalidator,
+		TxProcessor:    pr,
+		done:           make(chan struct{}),
+		MeshDB:         db,
+		config:         rewardConfig,
+		AtxDB:          atxDb,
+	}
+
+	latest, err := db.general.Get(LATEST)
```

Comment:
can't we just extract this to a function like Recover and call it instead of seperating constructurs ?

Reply:
i want to panic in the case of partial data retrieval, cant allow a recovered mesh and an unrecovered tortoise or vice versa and this was the easiest way of tying the two together

Reply:
maybe we can extract the differences to a new method `recover` anyway and let the new constructur call it.

Reply:
i extracted the common part between the constructors to avoid the duplication